modules/core/arrow-typeclasses/src/main/kotlin/arrow/typeclasses/MonadContinuations.kt

```diff
@@ -71,6 +71,7 @@ open class MonadContinuation<F, A>(M: Monad<F>, override val context: CoroutineC
       is BindingStrategy.Strict -> strategy.a
       is BindingStrategy.ContinuationShortCircuit -> suspendCoroutineUninterceptedOrReturn { c ->
```

Comment:
Isn't there a simpler API for this? Without `orReturn`

Reply:
It's used to short circuit the continuation. Could we keep track of it ourselves with a simple bool?

Reply:
I mean there should be another easier short in the API like `suspendCoroutineUnintercepted`

Reply:
Nope, such an api doesn't exist there. You're supposed to return `COROUTINE_SUSPENDED`. Actually, what is the difference between the 2 examples below? 🤔 The docs didn't make me any wiser, and I haven't dug into the code yet.

```
suspendCoroutineUninterceptedOrReturn { c ->
  c.resumeWithException(throw RuntimeException("Boom!"))
  COROUTINE_SUSPENDED
}
```

and

```
suspendCoroutine { c ->
  c.resumeWithException(throw RuntimeException("Boom!"))
}
```

Reply:
Beats me. Keep it as it is I guess. There's no interception in Unintercepted, that much we know...probably.